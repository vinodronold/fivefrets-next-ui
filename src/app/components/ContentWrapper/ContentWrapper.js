import React, { PureComponent } from 'react'
import { bool, node } from 'prop-types'
import { color } from '../../constants/theme'

export default class ContentWrapper extends PureComponent {
  static propTypes = {
    isMenuOpen: bool.isRequired,
    children: node.isRequired
  }
  render() {
    const { isMenuOpen, children } = this.props
    return (
      <div>
        {children}
        <style jsx>{`
          div {
            opacity: ${isMenuOpen ? 0 : 1};
            z-index: ${isMenuOpen ? 0 : 10};
            color: ${color.primary()};
          }
        `}</style>
      </div>
    )
  }
}
