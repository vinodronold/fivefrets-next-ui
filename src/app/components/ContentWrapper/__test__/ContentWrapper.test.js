import React from 'react'
import renderer from 'react-test-renderer'
import ContentWrapper from '../ContentWrapper'

describe('components/ContentWrapper', () => {
  it('should match menu is open', () => {
    const _content_open = renderer.create(<ContentWrapper isMenuOpen={true} />)
    expect(_content_open).toMatchSnapshot()
  })
  it('should match menu is closed', () => {
    const _content_closed = renderer.create(<ContentWrapper isMenuOpen={false} />)
    expect(_content_closed).toMatchSnapshot()
  })
})
