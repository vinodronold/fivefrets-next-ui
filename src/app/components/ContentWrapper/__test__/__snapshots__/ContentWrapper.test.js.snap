// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ContentWrapper should match menu is closed 1`] = `
<div
  className="jsx-2935080927"
/>
`;

exports[`components/ContentWrapper should match menu is open 1`] = `
<div
  className="jsx-115571663"
/>
`;
